nengo/utils/compat.py

```diff
@@ -30,6 +32,13 @@ def ensure_bytes(s):
         assert isinstance(s, bytes)
         return s
 
+    def replace(src, dst):
```

Comment:
Because `shutil.move` [depends on `os.rename` semantics](https://docs.python.org/2/library/shutil.html#shutil.move) and thus does not solve any of the problems. `os.replace` provides the correct semantics, but only exists in Python 3.3 and newer.